src/interpreter/crypto.rs

```diff
@@ -1,37 +1,35 @@
 use super::Interpreter;
-use crate::consts::{MEM_MAX_ACCESS_SIZE, VM_MAX_RAM};
-use crate::crypto;
+use crate::consts::{MEM_MAX_ACCESS_SIZE, MIN_VM_MAX_RAM_USIZE_MAX, VM_MAX_RAM};
 use crate::error::RuntimeError;
 
+use crate::arith::checked_add_word;
 use fuel_asm::PanicReason;
-use fuel_crypto::Hasher;
+use fuel_asm::PanicReason::ArithmeticOverflow;
+use fuel_crypto::{Hasher, Message, PublicKey, Signature};
 use fuel_types::{Bytes32, Bytes64, Word};
 
 impl<S> Interpreter<S> {
     pub(crate) fn ecrecover(&mut self, a: Word, b: Word, c: Word) -> Result<(), RuntimeError> {
-        if a > VM_MAX_RAM - Bytes64::LEN as Word
-            || b > VM_MAX_RAM - Bytes64::LEN as Word
-            || c > VM_MAX_RAM - Bytes32::LEN as Word
-        {
+        let bx = checked_add_word(b, Bytes64::LEN as Word)?;
+        let cx = checked_add_word(c, Bytes32::LEN as Word)?;
+
+        if a > VM_MAX_RAM - Bytes64::LEN as Word || bx > MIN_VM_MAX_RAM_USIZE_MAX || cx > MIN_VM_MAX_RAM_USIZE_MAX {
```

Comment:
Seems we might need a `pub const fn sub_usize(a: usize, b: usize) -> usize` in `arith` too so we can enable clippy unsafe arith check

Reply:
we already had a checked_sub_usize and I _just_ added it to this file